models/articles.js

```diff
@@ -50,10 +51,24 @@ exports.updateVotesOnArticleByID = (id, vote) => {
           return Promise.reject({ status: 404, msg: 'Article not found' });
         }
         return article.rows;
-      })
+      });
   }
   return Promise.reject({
     status: 400,
     msg: 'No votes submitted',
   });
 };
+
+exports.fetchArticleComments = async (id) => {
+  try {
+    const article = await this.fetchArticleByID(id);
+    if (article.status === 404) return Promise.reject(article);
+    const { rows } = await db.query(
+      `SELECT * FROM comments WHERE article_id = $1;`,
+      [id]
+    );
+    return rows;
+  } catch (err) {
+    return Promise.reject(err);
+  }
```

Comment:
if you have a catch block in your controller the error produced by a fail in this model should just be passed down the promise chain into that catch

aka you don't need a try catch block here as your catch in the controller will be passed this error by express

Reply:
Amazing!